lib/ain-grpc/src/rpc/eth.rs

```diff
@@ -846,7 +849,7 @@ impl MetachainRPCServer for MetachainRPCModule {
 
         while lo + 1 < hi {
             let sum = hi.checked_add(lo).ok_or(RPCError::ValueOverflow)?;
-            let mid = sum.checked_div(2u64).ok_or(RPCError::ValueOverflow)?;
+            let mid = sum.checked_div(2u64).ok_or(RPCError::DivideByZero)?;
```

Comment:
hmm.. its clearly state `divide by 2`.. should name the error not `DivideByZero`??
suggestion:
```suggestion
            let mid = sum.checked_div(2u64).ok_or(RPCError::ValueUnderflow)?;
```

Reply:
`checked_div` only throws when the divisor is 0.

Reply:
yea but the divisor has been hardcoded `2u64` right??

Reply:
@canonbrother's point is correct. The error type shouldn't be dividebyzero, but rather something more generic like, division error.

Reply:
Addressed in 9d2cd5cf2ab4a0975829a87b0d79b5f7f305bff0